operators/cmd/crownlabs-delete-labinstance/deleteCron.py

```diff
@@ -0,0 +1,173 @@
+import argparse
+import logging
+import re
+
+
+class InstanceExpiredDeleter:
+    """
+    This class interacts with crownlabs instances and templates to delete expired instances.
+    """
+    def __init__(self, dry_run=False):
+        """
+        Initializes the object.
+        :param dry_run: application run mode. Default value False.
+        """
+
+        self.dry_run = dry_run
+
+        try:
+            # Configuration loaded from a kube config
+            config.load_kube_config()
+        except config.config_exception.ConfigException:
+            # Configuration loaded from within a pod
+            config.load_incluster_config()
+
+    def _get_instance_list(self):
+        """
+        Gets the current list of instances.
+        :returns: the list of instances or None
+        """
+        # connect to Custom Object Api
+        api_instance = client.CustomObjectsApi()
+        try:
+            instances = api_instance.list_cluster_custom_object(group="crownlabs.polito.it", version="v1alpha2",
+                                                                plural="instances", pretty='pretty')
+        except ApiException as e:
+            logger.debug(f"Exception when calling CustomObjectsApi->list_cluster_custom_object: '{e}'")
+            return None
+        return instances
+
+    def instance_is_expired(self, time, timecreation):
+        """
+        Compares creation timestamp with current timestamp to decide if instance is expired.
+        :returns: True if expired or False if it is not
+        :time: instance life span in seconds
+        :timecreation: instance creation timestamp
+        """
+        now = datetime.now()
+        # calculate time difference to verify expiration status
+        deltatime = now-datetime.strptime(timecreation, '%Y-%m-%dT%H:%M:%SZ')
+        total_time = deltatime.days*24*60*60+deltatime.seconds
+        if(total_time > time):
+            # labinstance is expired
+            return True
+        else:
+            # labinstance not expired
+            return False
+
+    def _get_life_span(self, template_name, template_ns):
+        """
+        Retrieves deleteAfter field of template specified.
+        :returns: life span in seconds or None
+        :template_name: template name
+        :template_ns: template namespace
+        """
+        api_instance = client.CustomObjectsApi()
+        try:
+            template = api_instance.get_namespaced_custom_object(group="crownlabs.polito.it", version="v1alpha2",
+                                                                 namespace=template_ns, plural="templates",
+                                                                 name=template_name)
+        except ApiException as e:
+            logger.debug(f"Exception when calling CustomObjectsApi->get_namespaced_custom_object: '{e}'")
+            return None
+        delete_after = template.get("spec").get("deleteAfter")
+        temp = re.compile("([0-9]+)([mhd])")
```

Comment:
Suggestion: Isolate this important conversion logic in a dedicated function. This would make very easy to write a unit test to avoid wrong conversions.